### PR TITLE
Raise helpful error if `IOWebApp` fails to mount

### PR DESCRIPTION
**Pull request by gmkumar2005**
Error messages during initialization will help beginners falling into trouble.

### CODE REVIEW

**Comment by gmkumar2005** [comment]
With this change IOWebapp will throw an error if proper div is not found in the html which is used to load the scalajs application. 
`s"Unable to mount Calico. Check if $rootElementId exists in html")`

File: calico/src/main/scala/calico/IOWebApp.scala [hunk 1]
```diff
@@ -31,5 +31,14 @@ trait IOWebApp:
...
-    val rootElement = window.document.getElementById(rootElementId).map(_.get)
-    rootElement.flatMap(render.renderInto(_).useForever).unsafeRunAndForget()
+    window
+      .document
+      .getElementById(rootElementId)
+      .flatMap {
+        case Some(element) => IO.pure(element)
+        case None =>
+          IO.raiseError(new NoSuchElementException(
+            s"Unable to mount Calico. Check if $rootElementId exists in html"))
```
Comment (armanbilge) [inline calico/src/main/scala/calico/IOWebApp.scala:41]:
```suggestion
            s"Unable to mount ${getClass.getSimpleName.init} into element with id "$rootElementId", does it exist in the HTML?"))
```

Comment (gmkumar2005) [reply calico/src/main/scala/calico/IOWebApp.scala]: Done

**Comment by gmkumar2005** [comment]
Fixed the error message.

File: calico/src/main/scala/calico/IOWebApp.scala [hunk 2]
```diff
@@ -31,5 +31,14 @@ trait IOWebApp:
...
+    window
+      .document
+      .getElementById(rootElementId)
+      .flatMap {
+        case Some(element) => IO.pure(element)
+        case None =>
+          IO.raiseError(new NoSuchElementException(
+            s"Unable to mount ${getClass.getSimpleName.init} into element with id \"$rootElementId\", does it exist in the HTML?"))
+      }
+      .flatMap(render.renderInto(_).useForever)
```
Comment (armanbilge) [inline calico/src/main/scala/calico/IOWebApp.scala:43]:
```suggestion
      .flatMap {
        case Some(root) => render.renderInto(root).useForever
        case None =>
          IO.raiseError(new NoSuchElementException(
            s"Unable to mount ${getClass.getSimpleName.init} into element with id \"$rootElementId\", does it exist in the HTML?"))
      }
```

**Review by armanbilge — APPROVED** [review]
Thanks a lot! I can't believe I left a raw `map(_.get)` there 😅

**Pull request merged**